guice-extension/src/main/java/name/falgout/jeffrey/testing/junit/guice/GuiceExtension.java

```diff
@@ -79,10 +84,35 @@ private static Injector createInjector(ExtensionContext context)
       InvocationTargetException {
     Optional<Injector> parentInjector = getParentInjector(context);
     List<? extends Module> modules = getNewModules(context);
-
+    Set<Class<? extends Module>> moduleClasses = getContextModuleTypes(context);
+    if (isSharedInjector(context)) {
+      return createCachedInjector(modules, parentInjector, moduleClasses);
+    }
     return parentInjector
-        .map(injector -> injector.createChildInjector(modules))
-        .orElseGet(() -> Guice.createInjector(modules));
+            .map(injector -> injector.createChildInjector(modules))
```

Comment:
Could you move the caching logic into getOrCreateInjector? I think that would make a little more sense. That way, the createInjector class will always end up creating an injector